### ISSUE TITLE
💥 Compatibility with RN 0.62

### ISSUE THREAD

**Issue by A-Tokyo** [segmentio/analytics-react-native#165]
Upgrading to rn 0.62 causes android to fail because of the analytics module.

We fixed this issue by adding this to settings.gradle
```
// By Us - @segment/analytics-react-native-mixpanel - to make @segment/analytics-react-native-mixpanel work with rn 0.62.2 - 0.63-rc0
include ':@segment_analytics-react-native'
project(':@segment_analytics-react-native').projectDir = new File(rootProject.projectDir, '../node_modules/@segment/analytics-react-native/android')
```

I think this should be either added to the docs, or fixed internally

Segment versions:
```
"@segment/analytics-react-native": "^1.1.0",
"@segment/analytics-react-native-mixpanel": "^1.1.0",
```

Thanks a lot!

**Comment by A-Tokyo** [comment]
UPDATE: 

Temporary fix:

1) edit settings.gradle and add:
```
// By Us - @segment/analytics-react-native-mixpanel - to make @segment/analytics-react-native-mixpanel work with rn 0.62.2 - 0.63-rc0
include ':@segment_analytics-react-native'
project(':@segment_analytics-react-native').projectDir = new File(rootProject.projectDir, '../node_modules/@segment/analytics-react-native/android')
```
2) Edit app/build.gradle and add: (Note that you need to add it only for the packages you use eg firebase and mixpanel in this case) -- Author: @dagi3d 
```
// By Us - byUs_FixSegmentLibs for segment analytics https://github.com/segmentio/analytics-react-native/issues/164
task byUs_FixSegmentLibs {
    ant.replaceregexp(match:':@segment_analytics-react-native', replace:':segment_analytics-react-native', flags:'g', byline:true) {
        fileset(dir: '../../node_modules/@segment/analytics-react-native-mixpanel/android', includes: 'build.gradle')
    }
    ant.replaceregexp(match:':@segment_analytics-react-native', replace:':segment_analytics-react-native', flags:'g', byline:true) {
        fileset(dir: '../../node_modules/@segment/analytics-react-native-firebase/android', includes: 'build.gradle')
    }
}
byUs_FixSegmentLibs;

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
```


Note: The `By Us` prefix is a highly opinionated prefix we use to easily detect what we add to the native files usually edited by RN.

**Comment by A-Tokyo** [comment]
UPDATE: 

Temporary fix:

1) edit settings.gradle and add:
```
// By Us - @segment/analytics-react-native-mixpanel - to make @segment/analytics-react-native-mixpanel work with rn 0.62.2 - 0.63-rc0
include ':@segment_analytics-react-native'
project(':@segment_analytics-react-native').projectDir = new File(rootProject.projectDir, '../node_modules/@segment/analytics-react-native/android')
```
2) Edit app/build.gradle and add: (Note that you need to add it only for the packages you use eg firebase and mixpanel in this case) -- Author: @dagi3d 
```
// By Us - byUs_FixSegmentLibs for segment analytics https://github.com/segmentio/analytics-react-native/issues/165
task byUs_FixSegmentLibs {
    ant.replaceregexp(match:':@segment_analytics-react-native', replace:':segment_analytics-react-native', flags:'g', byline:true) {
        fileset(dir: '../../node_modules/@segment/analytics-react-native-mixpanel/android', includes: 'build.gradle')
    }
    ant.replaceregexp(match:':@segment_analytics-react-native', replace:':segment_analytics-react-native', flags:'g', byline:true) {
        fileset(dir: '../../node_modules/@segment/analytics-react-native-firebase/android', includes: 'build.gradle')
    }
}
byUs_FixSegmentLibs;

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
```


Note: The `By Us` prefix is a highly opinionated prefix we use to easily detect what we add to the native files usually edited by RN.

**Comment by bsneed** [comment]
@A-Tokyo thanks so much for doing the leg work here!! I'll try and get a release out soon that addresses these 0.62 incompatibilities.

**Comment by A-Tokyo** [comment]
@bsneed It's my duty ( :, Looking forward for the release 🚀

**Comment by A-Tokyo** [comment]
Possible fix in next release: https://github.com/segmentio/analytics-react-native/pull/169

**Comment by bsneed** [comment]
Fixed in #169
